src/crl.rs

```diff
@@ -193,6 +259,41 @@ impl<'a> BorrowedCertRevocationList<'a> {
         Ok(crl)
     }
 
+    /// Convert the CRL to an [`OwnedCertRevocationList`]. This may error if any of the revoked
+    /// certificates in the CRL are malformed or contain unsupported features.
+    ///
+    /// This function is only available when the "alloc" feature is enabled.
+    #[cfg(feature = "alloc")]
+    pub fn to_owned(&self) -> Result<OwnedCertRevocationList, Error> {
+        // Parse and collect the CRL's revoked cert entries, ensuring there are no errors. With
+        // the full set in-hand, create a lookup map by serial number for fast revocation checking.
+        let mut revoked_certs = HashMap::new();
+        self.into_iter()
```

Comment:
I think you should be able to `collect()` into `Result<HashMap<_, _>, _>` directly, by mapping the iterator to a `Result<(K, V), E>` tuple.

Reply:
Good call, thanks!